procbuild/listener.py

```diff
@@ -0,0 +1,186 @@
+import json
+import io
+import codecs
+import time
+import asyncio
+
+from multiprocessing import Process
+from concurrent.futures import ThreadPoolExecutor
+
+import zmq
+from zmq.asyncio import Context
+
+from . import MASTER_BRANCH
+from .message_proxy import OUT
+from .utils import file_age, log
+from .pr_list import get_pr_info, status_file, cache
+from .builder import BuildManager
+
+
+class Listener:
+    """ Listener class for defining zmq sockets and maintaining a build queue.
+    
+    Attributes:
+    ------------
+    ctx: zmq.asyncio.Context, the main context for the listener class
+    
+    prefix: str, the prefix listened for by zmq sockets
+    
+    socket: zmq.socket, the socket for listening to 
+    
+    queue: asyncio.Queue, the queue for holding the builds
+    
+    dont_build: set, unique collection of PRs currently in self.queue
+        Note: Only modify self.dont_build within synchronous blocks.
+    
+    """
+    
+    def __init__(self, prefix='build_queue'):
+        """
+        Parameters:
+        ------------
+        build_queue: str, the prefix that will be checked for by the zmq socket
+        """
+        self.ctx = Context.instance()
+        self.prefix = prefix
+
+        self.socket = self.ctx.socket(zmq.SUB)
+        self.socket.connect(OUT)
+        self.socket.setsockopt(zmq.SUBSCRIBE, self.prefix.encode('utf-8'))
+        
+        self.queue = asyncio.Queue()
+        self.dont_build = set()
+
+    async def listen(self):
+        """Listener method, containing while loop for checking socket
+        """
+        while True:
+            msg = await self.socket.recv_multipart()
+            target, raw_payload = msg
+            payload = json.loads(raw_payload.decode('utf-8'))
+            print('received', payload)
```

Comment:
Debug statement?

Reply:
Ok I'll remove it, I found it useful even when it was running to know that it was receiving things appropriately and it would show up in the heroku logs (I think). But I'm ok abandoning this.